it/controllers/registration/otherbusinessinvolvements/HaveVatNumberControllerISpec.scala

```diff
@@ -0,0 +1,165 @@
+        .registrationApi.getListSection[OtherBusinessInvolvement](None)
+
+      insertCurrentProfileIntoDb(currentProfile, sessionId)
+
+      val response: Future[WSResponse] = buildClient(pageUrl(idx1)).get()
+
+      whenReady(response) { res =>
+        res.status mustBe OK
+      }
+    }
+
+    "return OK with prepop" in new Setup {
+      implicit val s4lKey: S4LKey[OtherBusinessInvolvement] = OtherBusinessInvolvement.s4lKey(idx1)
+      given()
+        .audit.writesAudit()
+        .audit.writesAuditMerged()
+        .user.isAuthorised()
+        .registrationApi.getSection[EligibilitySubmissionData](Some(testEligibilitySubmissionData))
+        .s4lContainer[OtherBusinessInvolvement].isEmpty
+        .registrationApi.getSection[OtherBusinessInvolvement](Some(fullOtherBusinessInvolvement), idx = Some(idx1))
+
+      insertCurrentProfileIntoDb(currentProfile, sessionId)
+
+      val response: Future[WSResponse] = buildClient(pageUrl(idx1)).get()
+
+      whenReady(response) { res =>
+        res.status mustBe OK
+        Jsoup.parse(res.body).getElementById(HaveVatNumberForm.haveVatNumberKey).attr("value") mustBe testHasVrn.toString
+      }
+    }
+  }
+
+  s"GET ${pageUrl(idx2)}" must {
+    "return OK if it is a valid index" in new Setup {
+      implicit val s4lKey: S4LKey[OtherBusinessInvolvement] = OtherBusinessInvolvement.s4lKey(idx1)
+      given()
+        .audit.writesAudit()
+        .audit.writesAuditMerged()
+        .user.isAuthorised()
+        .registrationApi.getSection[EligibilitySubmissionData](Some(testEligibilitySubmissionData))
+        .s4lContainer[OtherBusinessInvolvement].isEmpty
+        .registrationApi.getSection[OtherBusinessInvolvement](None, idx = Some(idx1))
+        .registrationApi.getListSection[OtherBusinessInvolvement](Some(List(fullOtherBusinessInvolvement)))
+
+      insertCurrentProfileIntoDb(currentProfile, sessionId)
+
+      val response: Future[WSResponse] = buildClient(pageUrl(idx1)).get()
+
+      whenReady(response) { res =>
+        res.status mustBe OK
+      }
+    }
+
+    "return a redirect to a valid index" in new Setup {
+      implicit val s4lKey: S4LKey[OtherBusinessInvolvement] = OtherBusinessInvolvement.s4lKey(idx2)
+      given()
+        .audit.writesAudit()
+        .audit.writesAuditMerged()
+        .user.isAuthorised()
+        .registrationApi.getSection[EligibilitySubmissionData](Some(testEligibilitySubmissionData))
+        .s4lContainer[OtherBusinessInvolvement].isEmpty
+        .registrationApi.getSection[OtherBusinessInvolvement](None, idx = Some(idx2))
+        .registrationApi.getListSection[OtherBusinessInvolvement](None)
+
+      insertCurrentProfileIntoDb(currentProfile, sessionId)
+
+      val response: Future[WSResponse] = buildClient(pageUrl(idx2)).get()
+
+      whenReady(response) { res =>
+        res.status mustBe SEE_OTHER
+        res.header(HeaderNames.LOCATION) mustBe Some(pageUrl(idx1))
+      }
+    }
+  }
+
+  s"POST ${pageUrl(idx1)}" must {
```

Comment:
A test is missing for when the user doesn't answer the question (i.e. BAD REQUEST test)